snes/dsp.c

```diff
@@ -633,13 +634,35 @@ void dsp_write(Dsp *dsp, uint8_t adr, uint8_t val) {
   dsp->ram[adr] = val;
 }
 
-void dsp_getSamples(Dsp* dsp, int16_t* sampleData, int samplesPerFrame) {
+int16_t saturated_add16(int16_t a, int16_t b) {
+  int16_t c = a + b;
+  if (((a ^ b) & SHRT_MIN) == 0) {
+    if ((c ^ a) & SHRT_MIN) {
+        c = (a < 0) ? SHRT_MIN : SHRT_MAX;
+    }
+  }
+  return c;
+}
+
+// The mono version of a 2-channel audio sample is the average of those 2 channel samples,
+// but with some clamping / saturated addition to avoid integer overflow.
+int16_t monoSample(int16_t leftChannelSample, int16_t rightChannelSample) {
+  return saturated_add16(leftChannelSample, rightChannelSample) / 2;

```

Comment:
(leftChannelSample + rightChannelSample) >> 1 can never overflow.. so saturated_add16 can be removed.

Reply:
did you mean (left>>1) + (right>>1)? If so, I agree.

Reply:
also, did you mean to close this?